include/golioth/client.h

```diff
@@ -49,14 +49,15 @@ enum golioth_content_type
 };
 
 /// Response status and CoAP class/code
-struct golioth_response
+struct golioth_response_codes
 {
     /// Status to indicate whether a response was received
     ///
     /// One of:
+    ///      GOLIOTH_OK (CoAP codes was in the success class: 2.XX)
+    ///      GOLITOH_ERR_COAP_RESPONSE (CoAP error; anything other than 2.xx. Check CoAP codes)
     ///      GOLIOTH_ERR_TIMEOUT (no response received from server)
-    ///      GOLIOTH_OK (2.XX)
-    ///      GOLIOTH_ERR_FAIL (anything other than 2.XX)
+    ///      GOLIOTH_ERR_FAIL (All other failures not related to a response)
     enum golioth_status status;
```

Comment:
I think we could make things even clearer by removing the golioth_status from this struct and passing it separately, so that this struct only contains CoAP response codes.

Reply:
Most of the confusion actually arises inside the SDK. The problem is that they're _not_ always tightly coupled. When the `status` field holds a value that is neither `GOLIOTH_OK` nor `GOLIOTH_ERR_COAP_RESPONSE`, they're not coupled at all.